test/AstAnalyser.spec.js

```diff
@@ -254,30 +254,115 @@ describe("AstAnalyser", (t) => {
     });
   });
 
-  it("remove the packageName from the dependencies list", async () => {
-    const result = await getAnalyser().analyseFile(
-      new URL("depName.js", FIXTURE_URL),
-      { module: false, packageName: "foobar" }
-    );
-
-    assert.ok(result.ok);
-    assert.strictEqual(result.warnings.length, 0);
-    assert.deepEqual([...result.dependencies.keys()],
-      ["open"]
-    );
-  });
+  describe("analyseFile", () => {
+    it("remove the packageName from the dependencies list", async () => {
+      const result = await getAnalyser().analyseFile(
+        new URL("depName.js", FIXTURE_URL),
+        { module: false, packageName: "foobar" }
+      );
+
+      assert.ok(result.ok);
+      assert.strictEqual(result.warnings.length, 0);
+      assert.deepEqual([...result.dependencies.keys()],
+        ["open"]
+      );
+    });
+
+    it("should fail with a parsing error", async () => {
+      const result = await getAnalyser().analyseFile(
+        new URL("parsingError.js", FIXTURE_URL),
+        { module: false, packageName: "foobar" }
+      );
+
+      assert.strictEqual(result.ok, false);
+      assert.strictEqual(result.warnings.length, 1);
+
+      const parsingError = result.warnings[0];
+      assert.strictEqual(parsingError.kind, "parsing-error");
+    });
+
+    describe("hooks", () => {
+      const analyser = new AstAnalyser();
+      const url = new URL("depName.js", FIXTURE_URL);
+
+      describe("initialize", () => {
+        it("should throw if initialize is not a function", async () => {
+          const res = await analyser.analyseFile(
+            url, {
+            initialize: "foo"
+          });
 
-  it("should fail with a parsing error", async () => {
-    const result = await getAnalyser().analyseFile(
-      new URL("parsingError.js", FIXTURE_URL),
-      { module: false, packageName: "foobar" }
-    );
+          assert.strictEqual(res.ok, false);
+          assert.strictEqual(res.warnings[0].value, "options.initialize must be a function");
+          assert.strictEqual(res.warnings[0].kind, "parsing-error");
```

Comment:
@fraxken that doesn't make sense to me, but it seems that every error catched by `analyseFile` are wrapped into this result object.

So either we let it like that, or I throw like for the `analyze` function, but we'll do the check twice in this case.